matrix_client/room.py

```diff
@@ -456,15 +456,17 @@ def update_aliases(self):
         """
         try:
             response = self.client.api.get_room_state(self.room_id)
-            for chunk in response:
-                if "content" in chunk and "aliases" in chunk["content"]:
-                    if chunk["content"]["aliases"] != self.aliases:
-                        self.aliases = chunk["content"]["aliases"]
-                        return True
-                    else:
-                        return False
         except MatrixRequestError:
             return False
+        self.aliases = []
+        changed = False
+        for chunk in response:
+            if "content" in chunk and "aliases" in chunk["content"]:
+                for alias in chunk["content"]["aliases"]:
+                    if alias not in self.aliases:
+                        self.aliases.append(alias)
+                        changed = True
```

Comment:
I don't think this works as expected. Since `self.aliases` will always be empty because of `self.aliases = []` above, `changed` will always be `True` as long as there is at least one alias, even if it's the same as before.